### PR TITLE
 24.04 HW

### PR DESCRIPTION
**Pull request by belikovanastasya**
I fixed all mistakes, check please. But I have some bugs in createUser page:
Warning: You tried to redirect to the same route you're currently on: "/createuser";
And when I go to the User Page the  fields in the form doesn't fill automaicly

### CODE REVIEW

File: src/components/CreateUser/CreateUser.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import './createUser.scss';
+import { createUser } from '../../servises/users';
+import { Form } from '../Form';
+
+
+export const CreateUser = ({ history }) => {
+  const create = (user) => {
+    createUser(user)
+      .then((user) => {
+        history.push('/success');
```
Comment (iliatcymbal) [inline src/components/CreateUser/CreateUser.js]: argument user is useless here

File: src/components/Main/Invitation.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  }
+  render() {
+    return (
+      <div className="invitation">
+        {/* <p>Hello, { this.props.user.firstName}</p> */}
+        <div>You have {this.state.total} tasks</div>
+        <div>Done: {this.state.done}</div>
+        <div>In Progress: {this.state.inProgress}</div>
+        <div>Waiting: {this.state.waiting}</div>
+        <a href="task.html">Go to the task list</a>
```
Comment (iliatcymbal) [inline src/components/Main/Invitation.js]: it is not a valid href

File: src/components/Task/Task.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import './task.scss';
+import { getTask, updateTask, createTask } from '../../servises/tasks';
+
+
+export class Task extends Component {
+  constructor(props) {
+    super(props);
+
+    this.days = ['Вс', 'Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб'];
```
Comment (iliatcymbal) [inline src/components/Task/Task.js]: days should be in constants

File: src/components/TaskList/TaskList.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
+import { Link } from 'react-router-dom';
+import { Tabs, Tab } from '../Tabs';
+import { getTasks } from '../../servises/tasks';
+
+
+const days = ['Вс', 'Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб'];
```
Comment (iliatcymbal) [inline src/components/TaskList/TaskList.js]: should be moved to commom module (i.e. constants.js)

File: src/components/TaskList/TaskList.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+export class TaskList extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      tasksInWeek: []
+    };
+  }
+  componentDidMount() {
+    getTasks()
+      .then(data => this.setState({ tasksInWeek: data }));
```
Comment (iliatcymbal) [inline src/components/TaskList/TaskList.js]: it is not necessary, but can be simplified:
```
.then(tasksInWeek => this.setState({ tasksInWeek }));
```

File: src/components/TaskList/TaskList.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  createNewTask = (day) => {
+    this.props.history.push(`tasklist/newtask?day=${day}`);
+  };
+  getDay = () => {
+    return this.props.location.search.replase(/\D+/, '');
+  };
+  createNewTask = (day) => {
+    this.props.history.push(`tasklist/newtask?day=${day}`);
+  };
+  getDay = () => {
```
Comment (iliatcymbal) [inline src/components/TaskList/TaskList.js]: You have two `getDay` methods. And it is not used here

File: src/pages/Autorised.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { Route, Switch, Redirect } from 'react-router-dom';
+import { TaskList, Task, Main } from '../components';
+import { User, NotFound, Gallery } from '../pages';
+
+
+export const Autorised = ({ user }) => {
```
Comment (iliatcymbal) [inline src/pages/Autorised.js]: aut**H**orised

File: src/pages/Autorised.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import { Route, Switch, Redirect } from 'react-router-dom';
+import { TaskList, Task, Main } from '../components';
```
Comment (iliatcymbal) [inline src/pages/Autorised.js]: why are the pages in the components folder? If they are pages, they should be in pages folder

File: src/pages/Autorised.js [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+
+export const Autorised = ({ user }) => {
+  return (
+    <Switch>
+      <Route path="/" exact component={Main} />
+      <Route
+        path="/home"
+        exact
+        render={props => <Main {...props} user={user} />}
```
Comment (iliatcymbal) [inline src/pages/Autorised.js]: why you pass there _user_ to Main and don't pass it for `<Route path="/" exact component={Main} />`?

File: src/pages/Login/Login.js [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      showLoader: false
+    };
+  }
+  submit = (e) => {
+    const { email, password } = e.target;
+    e.preventDefault();
+    this.setState({ showLoader: true });
```
Comment (iliatcymbal) [inline src/pages/Login/Login.js:24]: so, when loader will be hidden? Try to enter wrong login credentials and check what happens

File: src/pages/Pages.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
+import { Autorised } from './Autorised';
+import { NonAuthorised } from './NonAuthorised';
```
Comment (iliatcymbal) [inline src/pages/Pages.js]: you don't have NonAuthorised.js in your repository

File: src/parts/Navigation/Navigation.js [hunk 12]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+import './navigation.scss';
+
+const navList = ['Home', 'Contacts', 'TaskList'];
+
+export const Navigation = ({ user }) => (
+  <nav>
+    <ul>
+      <li>{user && <a href="/user">{user.firstName}</a>}</li>
```
Comment (iliatcymbal) [inline src/parts/Navigation/Navigation.js]: when I click this link browser is reloaded. It shouldn't.

File: src/servises/users.js [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+
+export const login = (data = {}) => rest.post('public/login', data);
+
+export const logOut = () => rest.get('logout');
+
+export const checkUser = () => rest.get('public/checkUser');
+
+export const userInfo = () => rest.get('info');
+
+export const createUser = ({ firstName, lastName, email, password }) => rest.post('public/user', { firstName, lastName, email, password });
```
Comment (iliatcymbal) [inline src/servises/users.js]: why not just
```
export const createUser = (data) => rest.post('public/user', data);
```

The same for _updateUser_

**Review by iliatcymbal — COMMENTED** [review]
I have an error while building http://joxi.ru/BA0v7MBfJeDdbm

Please see my other notes and let me know when will be fixed (post comment in your git)

**Comment by belikovanastasya** [comment]
@iliatcymbal, check please

File: src/parts/Navigation/Navigation.js [hunk 14]
```diff
@@ -7,7 +7,8 @@ const navList = ['Home', 'Contacts', 'TaskList'];
 export const Navigation = ({ user }) => (
   <nav>
     <ul>
-      <li>{user && <a href="/user">{user.firstName}</a>}</li>
+      <li>{user && < NavLink to={`/user`}
+          exact >{user.firstName}</NavLink>}</li>
```
Comment (iliatcymbal) [inline src/parts/Navigation/Navigation.js]: no need exact here

**Review by iliatcymbal — APPROVED** [review]
looks good

**Pull request merged**